src/utils/RouteUtils.js

```diff
@@ -89,6 +89,38 @@ async function createSectionedRoutes(
   return sectionedRoutes;
 }
 
+/**
+ * Queries Graphhopper and returns exactly one walking route and any bus routes.
+ *
+ * The routes are processed prior to being returned. Note that GraphHopper always returns a walking route.
+ *
+ * @param destinationName
+ * @param end
+ * @param start
+ * @param departureTimeQuery
+ * @param isArriveBy
+ * @returns {Promise<Object>}
+ */
+async function getParsedWalkingAndBusRoutes(
+  destinationName: string,
+  end: string,
+  start: string,
+  departureTimeQuery: number,
+  isArriveBy: boolean,
+): Promise<{ parsedBusRoutes: ?Array<Object>, parsedWalkingRoute: Object }> {
+  const { busRoutes, walkingRoute } = await GraphhopperUtils.fetchRoutes(end, start, departureTimeQuery, isArriveBy);
+  const parsedWalkingRoute = ParseRouteUtils.parseWalkingRoute(
+    walkingRoute,
+    GraphhopperUtils.getDepartureTime(departureTimeQuery, isArriveBy, 0),
+    destinationName,
+  );
+
+  if (!busRoutes) return { parsedWalkingRoute, parsedBusRoutes: busRoutes };
```

Comment:
would it be clearer to just `return { parsedWalkingRoute, parsedBusRoutes: null }`